### PR TITLE
Fix install update

### PR DESCRIPTION
**Pull request by CPatchane**
* Fix application update in install modal
* Correctly remove the available version on update
* Some improvements and tests added

### CODE REVIEW

**Comment by CPatchane** [comment]
[[STO] Je ne peux pas mettre à jour un connecteur manuellement](https://trello.com/c/V2OiF66i/24-sto-je-ne-peux-pas-mettre-%C3%A0-jour-un-connecteur-manuellement)

File: src/ducks/apps/components/InstallModal.jsx [hunk 1]
```diff
@@ -9,10 +9,19 @@ import { hasPendingUpdate } from 'ducks/apps/appStatus'
...
-    const { app, onAlreadyInstalled } = this.props
-
     this.state = { activeTrap: true }
+  }
+
+  componentDidMount = () => {
+    this.handleInstalledStatus()
+  }
+
+  componentDidUpdate = () => {
```
Comment (gregorylegarec) [inline src/ducks/apps/components/InstallModal.jsx]: When does this modal update ?

File: src/ducks/apps/components/ApplicationRouting/ChannelRoute.jsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ export const ChannelRoute = ({
           isAppFetching={isAppFetching}
           isInstalling={isInstalling}
           onCurrentChannel={() => redirectTo(appPath)}
-          onNotInstalled={() => redirectTo(appPath)}
+          onNotHandled={() => redirectTo(appPath)}
```
Comment (gregorylegarec) [inline src/ducks/apps/components/ApplicationRouting/ChannelRoute.jsx]: What does `onNotHandled` mean ?

Comment (CPatchane) [reply src/ducks/apps/components/ApplicationRouting/ChannelRoute.jsx]: For the case where the channel modal doesn't not handle this application according to the app state/status

File: src/ducks/apps/components/InstallModal.jsx (hunk 1 above)
Comment (CPatchane) [reply src/ducks/apps/components/InstallModal.jsx]: After an install, the modal can be udpdated and we was checking the app status only in the constructor

File: src/ducks/apps/components/ApplicationRouting/ChannelRoute.jsx (hunk 2 above)
Comment (gregorylegarec) [reply src/ducks/apps/components/ApplicationRouting/ChannelRoute.jsx]: It seems that we should not even get there, should we ?

Comment (CPatchane) [reply src/ducks/apps/components/ApplicationRouting/ChannelRoute.jsx]: Since we don't check those statuses in the route actually we can get there

**Pull request merged**